apps/root/src/common/components/wallet-selector/index.tsx

```diff
@@ -105,6 +117,33 @@ const WalletSelector = ({ options, size = 'small' }: WalletSelectorProps) => {
     dispatch(cleanBalances());
   };
 
+  React.useEffect(() => {
+    const reFetchWalletsData = async () => {
```

Comment:
wouldn't this also trigger the first time the user logs in?

Reply:
It should not due to this condition:
`(prevWallets && prevWallets.length > 0 && wallets.length > prevWallets.length)`
We only update if there is already one wallet, and in this render we have more wallets than before